### PR TITLE
[Discussion] Cryotube Indicator

### PR DESCRIPTION
**Pull request by Kurfursten**
This makes the tube ding like a microwave every time it processes if the person inside is at full health.

### CODE REVIEW

**Comment by Duny-** [comment]
Sounds terrifying

**Comment by 9600bauds** [comment]
Now after someone dies in a plasma fire, after cloning them you can make a joke about "ding, the fries are ready!"

**Comment by d3athrow** [comment]
Add a cooldown for the sound.

**Comment by 9600bauds** [comment]
Does it ding once when their health is full for the first time, or does it ding constantly every 5 ticks for as long as their health is full?
I could see people getting annoyed at the second

**Comment by Duny-** [comment]
The later

**Comment by Kurfursten** [comment]
Duny's right.

**Comment by Iamgoofball** [comment]
why not just make it spit them out if they are at full HP

**Comment by Kurfursten** [comment]
It's just a random feature request from the thread. I suppose it's the same reason microwaves ding.

**Comment by 9600bauds** [comment]
Because they may hit full HP while still being full of toxins and thus die helpless on the ground.

**Comment by Probe1** [comment]
I still think this is for casual fags.

>why not just make it spit them out if they are at full HP?

absolutely disgusting.

**Comment by Kurfursten** [comment]
@Probe1 I could be mistaken, but I assumed the point was that the ding is for the benefit of the patient, not the doctor.

**Comment by d3athrow** [comment]
>why not just make it spit them out if they are at full HP

Cryo can keep somebody with internal bleeding alive, if they get spat out they'll die.

File: code/game/machinery/cryo.dm [hunk 1]
```diff
@@ -98,6 +99,12 @@
 		return
 
 	if(occupant)
+		if(occupant.health == occupant.maxHealth) //Is the occupant at full health? If so, ding!
+			if(cooldown == 0)
```
Comment (d3athrow) [inline code/game/machinery/cryo.dm]:
~~~
if(cooldown <= world.time)
    cooldown = world.time + (5 SECONDS)
    playsound(blahblah)
~~~

**Comment by Hoshi-chan** [comment]
It needs a sound other than the 'ding' that microwaves make, because that same sound is also used for the desk bells, which notify medbay personnel that someone is in the lobby. I guarantee that'll start to get confusing.

On top of that, that sound is too harsh to hear every 5 seconds. I'll look around, but you want something much softer and more ambient, not sharp and annoying.

**Comment by d3athrow** [comment]
maybe a soft ding-dong?

**Comment by Kurfursten** [comment]
@d3athrow Sorry, we had an IRC discussion so it didn't get posted here. We finally agreed to a visual change instead of auditory. Last I heard, Skowron is working on sprites.

**Comment by d3athrow** [comment]
waiting

**Comment by Heredth** [comment]
PLEASE YES +1

**Comment by d3athrow** [comment]
A visual indicator was added by me via #b7081790c4b29efaff2303810955a04d89c6224e

**Comment by Kurfursten** [comment]
Looks good

**Pull request closed without merging**